uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/contract-entitlements.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/absence-tab/modules/components',
+  'common/models/contract'
+], function (_, moment, components) {
+  components.component('contractEntitlements', {
+    bindings: {
+      contactId: '<'
+    },
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/contract-entitlements.html';
+    }],
+    controllerAs: 'entitlements',
+    controller: ['$controller', '$log', '$q', '$rootScope', 'AbsenceType', 'Contract', controller]
+  });
+
+  function controller ($controller, $log, $q, $rootScope, AbsenceType, Contract) {
+    $log.debug('Component: contract-entitlements');
+
+    var vm = this;
+
+    vm.contracts = null;
```

Comment:
I disagree with that. Please see my comment above https://github.com/civicrm/civihr/pull/1942#discussion_r122690782. I think null, [] and [...] allows you to store the state for the object in itself:

```js
null // not ready yet
[] // ready, but empty
[...] // ready and contains stuff 
```